nep-12.mediawiki

```diff
@@ -0,0 +1,2098 @@
+  console.log('Provider dAPI version: ' + version);
+  console.log('Provider dAPI compatibility: ' + JSON.stringify(compatibility));
+  console.log('Extra provider specific atributes: ' + JSON.stringify(compatibility));
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+    case CONNECTION_DENIED:
+      console.log('The user rejected the request to connect with your dApp.');
+      break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request, return:
+* The name of the provider wallet
+* The provider website
+* The provider specific version of the wallet
+* A list of NEP for which the specific provider protocol is compatible with
+
+Example
+<pre>
+{
+  name: 'Awesome Wallet',
+  website: 'https://www.awesome.com',
+  version: 'v0.0.1',
+  compatibility: [
+    'NEP-14',
+    'NEP-23',
+    'NEP-29'
+  ],
+  extra: {
+    theme: 'Dark Mode'
+  }
+}
+</pre>
+
+
+====getNetworks====
+Returns the networks the wallet provider has available to connect to, along with the default network the wallet is currently set to.
+
+
+'''Method Interface'''
+
+<pre>
+function getNetworks(): Promise<Networks>
+</pre>
+
+
+'''Input arguments'''
+
+None
+
+
+'''Success return value'''
+
+<pre>
+interface Networks {
+  networks: string[]; // Array of network names the wallet provider has available for the dapp developer to connect to.
+  defaultNetwork: string; // Network the wallet is currently set to.
+}
+</pre>
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string; // `NO_PROVIDER`|`CONNECTION_DENIED`
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+getNetworks()
+.then(response => {
+  const {
+    networks,
+    defaultNetwork,
+  } = response.networks;
+
+  console.log('Networks: ' + networks);
+  // eg. ["MainNet", "TestNet", "PrivateNet"]
+
+  console.log('Default network: ' + defaultNetwork);
+  // eg. "MainNet"
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+    case CONNECTION_DENIED:
+      console.log('The user rejected the request to connect with your dApp');
+      break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request, return:
+- The list of NEO networks that the wallet is able to connect to. The return values will be used by the dapp developer to communicate to which network they would like their request to be directed to. The wallet provider will be responsible for the logistics of which node to submit any request to for each network alias provided.
+- The wallet provider is to also return the default network that the wallet UI is currently set to by user.
+
+Example
+<pre>
+{
+  networks: ["MainNet", "TestNet", "PrivateNet"],
+  defaultNetwork: "TestNet",
```

Comment:
Should this call return the `defaultNetwork` plus the `currentNetwork`?

Reply:
Because that would save space for all other calls, which uses default or specified.

I do not know. It is just an idea.

Reply:
Yes, `currentNetwork` is also a good name for this. However, seems a bit late in the proposal process to change this now, as many apps and providers are already using `defaultNetwork`. Not sure it's worth the effort to lobby all parties to update. In addition to this, the wallet will also notify dapps via the `NETWORK_CHANGED` event in the case that the wallet network changes.